src/spikeinterface/core/numpyextractors.py

```diff
@@ -238,26 +305,94 @@ def from_peaks(peaks, sampling_frequency) -> "NumpySorting":
         sorting
             The NumpySorting object
         """
-        return NumpySorting.from_times_labels(peaks["sample_index"], peaks["channel_index"], sampling_frequency)
+        spikes = np.zeros(peaks.size, dtype=minimum_spike_dtype)
+        spikes["sample_index"] = peaks["sample_index"]
+        spikes["unit_index"] = peaks["channel_index"]
+        spikes["segment_index"] = peaks["segment_index"]
+
+        if unit_ids is None:
+            unit_ids = np.unique(peaks["channel_index"])
+
+        sorting = NumpySorting(spikes, sampling_frequency, unit_ids)
+
+        return sorting
 
 
 class NumpySortingSegment(BaseSortingSegment):
-    def __init__(self, units_dict):
+    def __init__(self, spikes, segment_index, unit_ids):
         BaseSortingSegment.__init__(self)
-        for unit_id, times in units_dict.items():
-            assert times.dtype.kind == "i", "numpy array of spike times must be integer"
-            assert np.all(np.diff(times) >= 0), "unsorted times"
-        self._units_dict = units_dict
+        self.spikes = spikes
+        self.segment_index = segment_index
+        self.unit_ids = list(unit_ids)
+        self.spikes_in_seg = None
 
     def get_unit_spike_train(self, unit_id, start_frame, end_frame):
-        times = self._units_dict[unit_id]
+        if self.spikes_in_seg is None:
+            # the slicing of segment is done only once the first time
+            # this fasten the constructor a lot
+            s0 = np.searchsorted(self.spikes["segment_index"], self.segment_index, side="left")
+            s1 = np.searchsorted(self.spikes["segment_index"], self.segment_index + 1, side="left")
+            self.spikes_in_seg = self.spikes[s0:s1]
+
+        unit_index = self.unit_ids.index(unit_id)
+
+        times = self.spikes_in_seg[self.spikes_in_seg["unit_index"] == unit_index]["sample_index"]
+
         if start_frame is not None:
             times = times[times >= start_frame]
         if end_frame is not None:
             times = times[times < end_frame]
         return times
 
 
+class SharedMemmorySorting(BaseSorting):
```

Comment:
Shouldn't it be `SharedMemorySorting`?

Reply:
Oups. thank you. Going back to my english class now.